cosmwasm/ibc-union/core/light-client-interface/src/lib.rs

```diff
@@ -170,6 +184,50 @@ fn client_impl<T: IbcClient>(
     Ok(addr)
 }
 
+pub struct StateUpdate<T: IbcClient> {
+    pub height: u64,
+    pub client_state: Option<T::ClientState>,
+    pub consensus_state: T::ConsensusState,
+    pub storage_writes: StorageWrites,
+}
+
+/// Client creation output type
+pub struct ClientCreation<T: IbcClient> {
+    /// The client state that is going to be stored by IBC. If set to `None`, IBC will store the
+    /// client state given by the creator as is
+    pub client_state: Option<T::ClientState>,
+    /// Custom events that will be emitted by IBC.
+    pub events: Vec<VerifyCreationResponseEvent>,
+    /// Arbitrary storage saves to the client's corresponding storage. These are accessible to the
+    /// client at any time.
+    pub storage_writes: StorageWrites,
+}
+
+impl<T: IbcClient> ClientCreation<T> {
+    pub fn empty() -> Self {
+        Self {
+            client_state: None,
+            events: Vec::new(),
+            storage_writes: Vec::new(),
+        }
+    }
+
+    pub fn add_event(mut self, event: VerifyCreationResponseEvent) -> Self {
+        self.events.push(event);
+        self
+    }
+
+    pub fn add_storage_write(mut self, kv: (Bytes, Bytes)) -> Self {
```

Comment:
since we require `T: Decode<Self::Encoding>`, why doesn't this use `T: Encode<Self::Encoding>`?

Reply:
you are correct, it would be an easier API